tools/osbuilder/dockerfiles/QAT/run.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+set -u
+
+# NOTE: Some env variables are set in the Dockerfile - those that are
+# intended to be over-rideable.
+export QAT_SRC=~/src/QAT
+export ROOTFS_DIR=~/src/rootfs
+export GOPATH=~/src/go
+export PATH=${PATH}:/usr/local/go/bin:${GOPATH}/bin
+
+katarepo=github.com/kata-containers/kata-containers
+katarepopath=${GOPATH}/src/${katarepo}
+
+testsrepo=github.com/kata-containers/tests
+testsrepopath=${GOPATH}/src/${testsrepo}
+
+grab_kata_repos()
+{
+    # Check out all the repos we will use now, so we can try and ensure they use the specified branch
+    # Only check out the branch needed, and make it shallow and thus space/bandwidth efficient
+    # Use a green prompt with white text for easy viewing
+    bin/echo -e "\n\e[1;42mClone and checkout Kata repos\e[0m" 
+    git clone --single-branch --branch $KATA_REPO_VERSION --depth=1 https://${katarepo} ${katarepopath}
+    git clone --single-branch --branch $KATA_REPO_VERSION --depth=1 https://${testsrepo} ${testsrepopath}
+}
+
+configure_kernel()
+{
+    cp /input/qat.conf ${katarepopath}/tools/packaging/kernel/configs/fragments/common/qat.conf
+    # We need yq and go to grab kernel versions etc.
+    ${testsrepopath}/.ci/install_yq.sh
+    ${testsrepopath}/.ci/install_go.sh -p
+    cd ${katarepopath}
+    /bin/echo -e "\n\e[1;42mDownload and configure Kata kernel with CRYPTO support\e[0m"
+    ./tools/packaging/kernel/build-kernel.sh setup
+}
+
+build_kernel()
+{
+    cd ${katarepopath}
+    LINUX_VER=$(ls -d kata-linux-*)
+    sed -i 's/EXTRAVERSION =/EXTRAVERSION = .qat.container/' $LINUX_VER/Makefile
+    /bin/echo -e "\n\e[1;42mBuild Kata kernel with CRYPTO support\e[0m" 
+    ./tools/packaging/kernel/build-kernel.sh build
+}
+
+build_rootfs()
+{
+    # Due to an issue with debootstrap unmounting /proc when running in a
+    # --privileged container, change into /proc to keep it from being umounted. 
+    # This should only be done for Ubuntu and Debian based OS's. Other OS 
+    # distributions had issues if building the rootfs from /proc
```

Comment:
I wonder if there is an open Debian/Ubuntu bug we could reference here?

Reply:
I thought the same, but I don't know enough about how `debootstrap` works to know if this is actually a bug or not.  When running on a bare metal host everything works fine. When running without `--privileged` and just adding capabilities it also works without issue because `debootstrap` doesn't have enough permission to actually umount `/proc` .   I did find this https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921815 and think it might be the same thing. What are your thoughts?  Should I include this link in this script?  I'm not sure fixing it would necessarily fix the issue for me but it is the closest thing to what I think is causing this problem and needed workaround.